File: abc113/b.c
#include<stdio.h>
#include<math.h>

#define INF 99999999

int main()
{
    unsigned int N, T, H[100000], i, ans;
    signed int A;
    double min = INF, avg, diff;

    scanf("%d", &N);
    scanf("%d %d", &T, &A);

    for (i = 0; i < N; i++)
    {
        scanf("%d", &H[i]);
    }

    for (i = 0; i < N; i++)
    {
        avg = T - H[i] * 0.006;
        diff = fabs((double)A - avg);
        if (diff < min)
        {
            ans = i + 1;
            min = diff;
        }
    }

    printf("%d", ans);
}

File: abc113/b.py
N = int(input())
T,A = map(int, input().split())
H = [h for h in map(int, input().split())]

m = 100000
ans = 1
for i, h in enumerate(H, 1):
    a = abs(A-(T-h*0.006))
    m = min(m, a)
    if m == a:
        ans = i

print(ans)
